src/components/__tests__/RegisterChallengePayment.cy.js

```diff
@@ -0,0 +1,372 @@
+
+    // if company is paying the fee, user still has option to add donation
+    testDonation();
+  });
+
+  it('if selected company and no coordinator - allows to register as coordinator + donate option', () => {
+    /**
+     * Scenario: Selected company has no company coordinator
+     * Displays checkbox with coordinator option
+     * If option is checked, displays partial form for registration
+     * TODO: Add condition - NO COORDINATOR IN SELECTED COMPANY
+     */
+    cy.dataCy(selectorCompany).should('not.exist');
+    cy.dataCy(getRadioOption(optionCompany)).should('be.visible').click();
+    cy.dataCy(selectorCompany).should('be.visible');
+    // checkbox coordinator
+    cy.dataCy(selectorCoordinatorCheckbox)
+      .should('be.visible')
+      .and('have.css', 'font-size', '14px')
+      .and('have.css', 'font-weight', '700')
+      .and('have.color', primary)
+      .and(
+        'contain',
+        i18n.global.t('companyCoordinator.labelRegisterCoordinator'),
+      );
+    // only show form when enabled
+    cy.dataCy(selectorCoordinatorText).should('not.exist');
+    cy.dataCy(selectorCoordinatorPhone).should('not.exist');
+    cy.dataCy(selectorCoordinatorJobTitle).should('not.exist');
+    cy.dataCy(selectorCoordinatorResponsibility).should('not.exist');
+    cy.dataCy(selectorCoordinatorTerms).should('not.exist');
+    // enable
+    cy.dataCy(selectorCoordinatorCheckbox).click();
+    // form
+    cy.dataCy(selectorCoordinatorText)
+      .should('be.visible')
+      .and('have.css', 'font-size', '14px')
+      .then(($el) => {
+        cy.stripHtmlTags(
+          i18n.global.t('companyCoordinator.textBecomeCoordinator'),
+        ).then((translation) => {
+          expect($el.text()).to.eq(translation);
+        });
+      });
+    // phone label
+    cy.dataCy(selectorCoordinatorPhone).should('be.visible');
+    // input label
+    cy.dataCy(selectorCoordinatorJobTitle).should('be.visible');
+    // checkbox responsibility
+    cy.dataCy(selectorCoordinatorResponsibility).should('be.visible');
+    // checkbox terms
+    cy.dataCy(selectorCoordinatorTerms).should('be.visible');
+
+    // if coordinator user still has option to add donation
+    testDonation();
+  });
+
+  it.only('if selected school - renders school select + donate option', () => {
+    cy.dataCy(selectorCompany).should('not.exist');
+    cy.dataCy(getRadioOption(optionSchool)).should('be.visible').click();
+    cy.dataCy(selectorCompany).should('be.visible');
+
+    // if coordinator user still has option to add donation
+    testDonation();
+  });
+}
+
+function getRadioOption(value) {
+  return `radio-option-${value}`;
+}
+
+function testDonation() {
+  cy.dataCy(selectorDonationCheckbox).click();
+  cy.dataCy(selectorSliderNumberInput)
+    .should('be.visible')
+    .and('have.value', defaultPaymentAmountMin.toString());
+  // click in the middle of slider
+  cy.dataCy(selectorSliderNumberSlider).should('be.visible').click();
```

Comment:
Instead of click on the slider widget you can use following code example:

```ts
cy.get('input[type=range]')
  .invoke('val', 20001)
  .trigger('change')
```

https://stackoverflow.com/questions/55023428/how-do-interact-correctly-with-a-range-input-slider-in-cypress

Reply:
This StackOverflow answer seems to be outdated. Quasar slider does not contain any input element. The state seems to be hadled by vue.
Value can be read from aria-valuenow, but cannot be set by .invoke() or .trigger() methods.
I tried the `.trigger('input')` method mentioned in comments, but it did not work either.

Reply:
> This StackOverflow answer seems to be outdated. Quasar slider does not contain any input element. The state seems to be hadled by vue. Value can be read from aria-valuenow, but cannot be set by .invoke() or .trigger() methods. I tried the `.trigger('input')` method mentioned in comments, but it did not work either.

This is only general example (use mouse for setting slider position is not good idea). You can change Quasar Slider widget model value.

Reply:
I worked around this. Instead, I am testing that the component emits the correct value on change of input.
Tests for slider component are redundant, since it is the default Quasar component and should be tested within library tests.